### DISCUSSION TITLE
Assigning IPs to a Prefix during Bulk Import

### DISCUSSION THREAD

**Discussion by RSDeuce** [netbox-community/netbox#8283] [Q&A] [answered]
Hello all. I am coming into an existing Netbox IPAM. The people who have set this up normally add IP addresses on a one-by-one basis to prefixes that they have created. They select **IPAM > Prefixes**, browse to the prefix, click the IP Addresses tab and select **Add an IP Address**. This works fine. 

However, they are behind on multiple (hundreds) of IP addresses that have been created in recent months. I want to bulk import these IPs from CSV onces, with the long term goal of automating the import of new IPs via the API. 

My issue: During bulk import, assigning an IP to a Prefix was not available. Post import, in the edit screen I also cannot assign an IP to a Prefix that I can see. It seems like the only way to get an IP to appear within the Prefix IP tab is to manually create them in that via the GUI. 

Browsing the API and comparing an IP address created within the Prefix tab, and one created during bulk import I don't see that the Prefix assignment is exposed and 'changeable'. 

My question: Am I missing something, or is this a feature request? Is this instance managed 'correctly' or should I speak to the owners about reorganizing the data, and possibly rebuilding from scratch? 

I am brand new to Netbox, having only used phpIPAM at a previous position, which had a very different logical structure to it's GUI and API. Thanks for any help.

**Comment by kkthxbye-code** [comment, marked as answer]
> My issue: During bulk import, assigning an IP to a Prefix was not available. Post import, in the edit screen I also cannot assign an IP to a Prefix that I can see. It seems like the only way to get an IP to appear within the Prefix IP tab is to manually create them in that via the GUI.

IP addresses are not assigned to prefixes. The relationship is calculated automatically. So if you create a prefix 10.200.0.0/24 that has the vrf "test" and csv import the following:

```
address,status,vrf
10.200.0.1/24,active,test
10.200.0.2/24,active,test
10.200.0.3/24,active,test
10.200.0.4/24,active,test
```

These will all automatically show up under the prefix and they will all reference the created prefix as their parent.

Not sure what could be wrong, maybe you are not creating them with the right netmask or in the wrong VRF?

**Reply by RSDeuce** [reply]
Thanks for your answer. You were completely correct.

There is a configured and correct VRF for the IPs I was bulk-importing, but the Prefix itself was not in that VRF. Changing the Prefix brought those IPs in immediately.

The docs weren't clear on that automatic relationship (or quite possibly I missed that they told me) so thanks for the quick and concise answer.
